### DISCUSSION TITLE
ESS External Control & CCCM Parameters

### DISCUSSION THREAD

**Discussion by BipedalPrimate** [Louisvdw/dbus-serialbattery#791] [Q&A] [answered]
Congratulations to all that have contributed & created this software. Love the flexibility of what appears in the config.default.ini (config.ini) file.

Have been looking at it for use on a forthcoming installation but have a couple of questions.

1. Should I set the ESS Mode to External Control?

2. Is there a maximum number of what appear to be array elements in config.ini for setting the thresholds in each of the 3 CCL/DCL algorithms? 
   Currently, Discharge/Charge Cell Voltage have 4 each, Charge Temperature has 9, Discharge Temperature has 7 and Charge/Discharge SOC have 3 each.

**Comment by mr-manuel** [comment, marked as answer]
1. Does not depend on this driver, but on your planned use. If you want to control all ESS functions by your own via Node-RED for example, then you should change it. If you have no clue what this is, leave it as it is.

2. There is no limit, but the more elemnts you have in an array the worse is the performance of the driver and the higher the CPU usage.
